src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -156,33 +125,32 @@ public void Running_Timer_CanBeFinalizedAndStopsFiring()
     {
         AutoResetEvent are = new AutoResetEvent(false);
         TimerCallback tc = new TimerCallback((object o) => are.Set());
-        var t = new Timer(tc, null, 100, 500);
-        Assert.True(are.WaitOne(250), "Failed to get first timer fire");
+        var t = new Timer(tc, null, 1, 500);
+        Assert.True(are.WaitOne(MaxPositiveTimeoutInMs), "Failed to get first timer fire");
         t = null; // Remove our refence so the timer can be GC'd 
         GC.Collect();
         GC.WaitForPendingFinalizers();
         GC.Collect();
-        Assert.False(are.WaitOne(750), "Should not have received a timer fire after it was collected");
+        Assert.False(are.WaitOne(500), "Should not have received a timer fire after it was collected");
     }
 
     [Fact]
-    public void MultpleTimers_PeriodicTimerIsntBlockedByABlockedTimer()
+    public void MultpleTimers_PeriodicTimerIsntBlockedByBlockedCallback()
     {
-        AutoResetEvent are1 = new AutoResetEvent(false);
-        AutoResetEvent are2 = new AutoResetEvent(false);
-        TimerCallback tc = new TimerCallback((object o) =>
+        ManualResetEventSlim mres = new ManualResetEventSlim();
```

Comment:
I realize this was in my pseudo-code example, but it's not necessary (I think I was originally intending to use it to signal the calling thread, but I ended up using the same Barrier for that); you can delete this line, as the mres is never used.